crates/solver/src/settlement/settlement_encoder.rs

```diff
@@ -350,10 +353,21 @@ impl SettlementEncoder {
                     .checked_div(uniform_sell_price)
                     .context("sell_amount computation failed")?;
                 // We have to sell slightly more `sell_token` to capture the `surplus_fee`
-                let sell_amount_adjusted_for_fees = sell_amount
+                let sell_amount = sell_amount
                     .checked_add(fee)
                     .context("sell_amount computation failed")?;
-                (sell_amount_adjusted_for_fees, executed_amount)
+
+                // Increase the `sell_amount` by protocol fee
+                let surplus = order
+                    .data
+                    .sell_amount
+                    .checked_sub(sell_amount)
+                    .unwrap_or(0.into());
+                let protocol_fee = surplus / (U256::from_f64_lossy(1. / protocol_fee_factor));
+                let protocol_fee_cap = sell_amount / (U256::from_f64_lossy(1. / protocol_fee_cap));
```

Comment:
Using `sell_amount` and not the limit price since limit price can be very big/small.